### PR TITLE
Add onItemClick and setShowMenu props to ContextualMenu to support dismissal on item click

### PR DESCRIPTION
**Pull request by lenahong**
### Platforms Impacted
- [ ] iOS
- [ ] macOS
- [X] win32
- [X] windows
- [ ] android

### Description of changes

- Add onItemClick prop support to ContextualMenu
- Add setShowMenu prop to ContextualMenu to have it pass down to ContextualMenuItem to handle menu show on item click
- Add Callout Dependency to ContextualMenu package

### Verification

- ContextualMenu visibility indicator was added to Test Component page
- Last Item Clicked indicator was added to the parent surface of ContextualMenu to track item click 

### Pull request checklist

This PR has considered (when applicable):
- [ ] Automated Tests
- [X] Documentation and examples
- [ ] Keyboard Accessibility
- [ ] Voiceover
- [ ] Internationalization and Right-to-left Layouts

### CODE REVIEW

File: apps/fluent-tester/src/RNTester/TestComponents/ContextualMenu/ContextualMenuTest.tsx [hunk 1]
```diff
@@ -10,36 +10,63 @@ export const ContextualMenuTest: React.FunctionComponent<{}> = () => {
...
   return (
     <View>
       <Text style={fabricTesterStyles.testSection} testID={CONTEXTUALMENU_TESTPAGE}>
-        ContextualMenu Test
+        Standard ContextualMenu
       </Text>
       <Separator />
-      <Button content="Press for ContextualMenu" onClick={toggleShowContextualMenu} componentRef={stdBtnRef} />
+      <View style={{ flexDirection: 'column', paddingVertical: 5 }}>
+        <Text selectable={true}>
```
Comment (PPatBoyd) [inline apps/fluent-tester/src/RNTester/TestComponents/ContextualMenu/ContextualMenuTest.tsx:46]: could skip selectable

File: packages/components/Callout/src/Callout.tsx [hunk 2]
```diff
@@ -17,7 +17,7 @@ export const Callout = compose<ICalloutType>({
     const [targetNativeTag, setTargetNativeTag] = React.useState<number>(undefined);
 
     React.useLayoutEffect(() => {
-      if (target?.current) {
+      if (target ?.current) {
```
Comment (PPatBoyd) [inline packages/components/Callout/src/Callout.tsx:20]: ?

File: packages/components/ContextualMenu/src/ContextualMenu.types.ts [hunk 3]
```diff
@@ -1,7 +1,25 @@
...
+  /*
+   ** The currently selected ContextualMenu's key
+   */
+  selectedKey: string | null;
+
+  /*
+   ** Updates the clicked menu item and calls the client’s onItemClick callback
+   */
+  onItemClick?: (key: string) => void;
+  dismissMenu?: () => void;
```
Comment (PPatBoyd) [inline packages/components/ContextualMenu/src/ContextualMenu.types.ts:16]: onDismissMenu ?

File: packages/components/ContextualMenu/src/ContextualMenuItem.tsx [hunk 4]
```diff
@@ -22,25 +24,39 @@ export const ContextualMenuItem = compose<ContextualMenuItemType>({
...
-    // attach the pressable state handlers
-    const pressable = useAsPressable({ ...rest, onPress: onClick });
-    const onKeyUp = React.useCallback(
+    // Grabs the context information from ContextualMenu (currently selected menuItem and client's onItemClick callback)
+    const context = React.useContext(CMContext);
+
+    const onItemClick = React.useCallback(
       e => {
-        if (onClick && (e.nativeEvent.key === 'Enter' || e.nativeEvent.key === ' ')) {
+        context.dismissMenu && context.dismissMenu();
```
Comment (PPatBoyd) [inline packages/components/ContextualMenu/src/ContextualMenuItem.tsx:32]: could context?.dismissMenu?()  iirc might be ?.()

File: packages/components/ContextualMenu/src/ContextualMenuItem.types.ts [hunk 5]
```diff
@@ -53,7 +54,7 @@ export interface ContextualMenuItemProps extends Omit<IPressableProps, 'onPress'
 }
 
 export interface ContextualMenuItemSlotProps {
-  root: React.PropsWithRef<ViewProps>;
+  root: React.PropsWithRef<ViewProps> & { acceptsKeyboardFocus?: boolean };
```
Comment (PPatBoyd) [inline packages/components/ContextualMenu/src/ContextualMenuItem.types.ts:57]: is it legit to have a context menu that doesn't accept keyboard focus?  What's the scenario?

File: packages/components/Callout/src/Callout.tsx (hunk 2 above)
Comment (lenahong) [reply packages/components/Callout/src/Callout.tsx:20]: Not sure why this change was added. I'll unstage this

File: packages/components/ContextualMenu/src/ContextualMenuItem.types.ts (hunk 5 above)
Comment (lenahong) [reply packages/components/ContextualMenu/src/ContextualMenuItem.types.ts:57]: No we want keyboard focus! This was to test focus behaviors for my next work item. I"ll remove it

**Pull request merged**